src/coreclr/tools/aot/ILCompiler.TypeSystem.ReadyToRun.Tests/MarshalUtilsTests.cs

```diff
@@ -0,0 +1,71 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Internal.TypeSystem;
+using Internal.TypeSystem.Interop;
+
+using Xunit;
+
+namespace TypeSystemTests
+{
+    public class MarshalUtilsTests
+    {
+        private TestTypeSystemContext _context;
+        private ModuleDesc _testModule;
+
+        public MarshalUtilsTests()
+        {
+            _context = new TestTypeSystemContext(TargetArchitecture.X64);
+            var systemModule = _context.CreateModuleForSimpleName("CoreTestAssembly");
+            _context.SetSystemModule(systemModule);
+
+            _testModule = systemModule;
+        }
+
+        [Theory]
+        [InlineData(WellKnownType.Void)]
+        [InlineData(WellKnownType.Boolean)]
+        [InlineData(WellKnownType.Char)]
+        [InlineData(WellKnownType.SByte)]
+        [InlineData(WellKnownType.Byte)]
+        [InlineData(WellKnownType.Int16)]
+        [InlineData(WellKnownType.UInt16)]
+        [InlineData(WellKnownType.Int32)]
+        [InlineData(WellKnownType.UInt32)]
+        [InlineData(WellKnownType.Int64)]
+        [InlineData(WellKnownType.UInt64)]
+        [InlineData(WellKnownType.IntPtr)]
+        [InlineData(WellKnownType.UIntPtr)]
+        [InlineData(WellKnownType.Single)]
+        [InlineData(WellKnownType.Double)]
+        [InlineData(WellKnownType.Nullable)]
+        [InlineData(WellKnownType.RuntimeFieldHandle)]
+        [InlineData(WellKnownType.RuntimeTypeHandle)]
+        [InlineData(WellKnownType.RuntimeMethodHandle)]
+        [InlineData(WellKnownType.TypedReference)]
+        [InlineData(WellKnownType.ByReferenceOfT)]
+        public void IsBlittableType_BilittableWellKnownTypes_ReturnsTrue(WellKnownType type) =>
```

Comment:
Nullable, TypedReference and ByReferenceOfT should not be in this list.

- Nullable is an uninstantiated generic, and it doesn't make sense to ask this question.
- TypedReference and ByReferenceOfT should not be considered blittable at all as they hold byreference pointers which are not safe to marshal to native code under any circumstance.